src/hooks/useLoadLocalMessages.ts

```diff
@@ -0,0 +1,26 @@
+import { useCallback } from 'react'
+import { useStore } from '../components/Store'
+import { db } from '../utils/db'
+import { MessagePayload } from '../utils/types'
+
+type MessageLoader = () => Promise<MessagePayload[]>
+
+export default function useLoadLocalMessages(): MessageLoader {
```

Comment:
This does not have to be a hook, I think. Could be a regular utility/getter function like

```js
export default async function getLocalMessagesForRoom(roomId) {
    if (!roomId) {
        return []
    }

    const messages = await db.messages
        .where('roomId')
        .equals(roomId)
        .toArray()

    return messages.map((m) => JSON.parse(m.serialized) as MessagePayload)
}
```